src/Connector.AzureDataLake/Connector/AzureDataLakeConnector.cs

```diff
@@ -135,9 +350,86 @@ public override Task<IAsyncEnumerable<ConnectorLatestEntityPersistInfo>> GetLate
 
         public override async Task<ConnectionVerificationResult> VerifyConnection(ExecutionContext executionContext, IReadOnlyDictionary<string, object> config)
         {
-            await _client.EnsureDataLakeDirectoryExist(new AzureDataLakeConnectorJobData(config.ToDictionary(x => x.Key, x => x.Value)));
+            try
+            {
+                var jobData = await AzureDataLakeConnectorJobData.Create(executionContext, config.ToDictionary(config => config.Key, config => config.Value));
+                await _client.EnsureDataLakeDirectoryExist(jobData);
+
+                if (jobData.IsStreamCacheEnabled)
+                {
+                    if (string.IsNullOrWhiteSpace(jobData.StreamCacheConnectionString))
+                    {
+                        return new ConnectionVerificationResult(false, $"Stream cache connection string must be valid when buffer is enabled.");
+                    }
+                    await using var connection = new SqlConnection(jobData.StreamCacheConnectionString);
+                    await connection.OpenAsync();
```

Comment:
If we associate a transaction with the connection then we don't need to delete the test table later

Reply:
as discussed, we put this inside transactions so that we can test delete too